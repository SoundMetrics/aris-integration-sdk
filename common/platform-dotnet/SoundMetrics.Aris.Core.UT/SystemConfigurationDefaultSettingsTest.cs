using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Core.ApprovalTests;

namespace SoundMetrics.Aris.Core
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class SystemConfigurationDefaultSettingsTest
    {
        private static readonly ObservedConditions observedConditions = new((Temperature)15, (Distance)5);
        private static readonly Salinity salinity = Salinity.Fresh;

        [TestMethod]
        public void Aris3000DefaultSettings()
        {
            RunTest(nameof(Aris3000DefaultSettings), SystemType.Aris3000);
        }

        [TestMethod]
        public void Aris1800DefaultSettings()
        {
            RunTest(nameof(Aris1800DefaultSettings), SystemType.Aris1800);
        }

        [TestMethod]
        public void Aris1200DefaultSettings()
        {
            RunTest(nameof(Aris1200DefaultSettings), SystemType.Aris1200);
        }

        private static void RunTest(string testName, SystemType systemType)
        {
            var helper = new PrettyPrintHelper(0);
            helper.PrintHeading(testName);

            var systemCfg = SystemConfiguration.GetConfiguration(systemType);

            helper.PrintHeading("Inputs");

            using (var _ = helper.PushIndent())
            {
                helper.PrintValue(nameof(observedConditions), observedConditions);
                helper.PrintValue(nameof(salinity), salinity);
            }

            var defaultSettings =
                systemCfg.GetDefaultSettings(observedConditions, salinity, out var windowBounds);

            helper.PrintHeading($"Default settings for [{systemType}]");
            using (var _ = helper.PushIndent())
            {
                helper.PrintValue(nameof(windowBounds), windowBounds);
                helper.PrintValue(nameof(defaultSettings), defaultSettings);
            }

            Approvals.Verify(helper.ToString());
        }
    }
}
