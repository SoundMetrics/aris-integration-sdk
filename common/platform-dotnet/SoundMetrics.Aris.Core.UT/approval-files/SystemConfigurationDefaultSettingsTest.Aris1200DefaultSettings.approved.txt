Aris1200DefaultSettings: 
Inputs: 
    observedConditions: ObservedConditions: 
        WaterTemp: 15.000 °C
        Depth: 5.000 m
    salinity: Fresh
Default settings for [ARIS 1200 (2)]: 
    windowBounds: WindowStart=[4.000 m]; WindowEnd=[40.006 m]; WindowLength=[36.006 m]
    defaultSettings: AcousticSettingsRaw: 
        SystemType: ARIS 1200 (2)
        FrameRate: 5.942194333523484/s
        SampleCount: 2456
        SampleStartDelay: 0.005457 s
        SamplePeriod: 0.000020 s
        PulseWidth: 0.000040 s
        PingMode: PingMode 1, 48 beams, 3 pings per frame
        EnableTransmit: True
        Frequency: Low
        Enable150Volts: True
        ReceiverGain: 20
        FocusDistance: 22.003 m
        AntiAliasing: 0.000000 s
        InterpacketDelay: InterpacketDelaySettings: 
            Enable: False
            Delay: 0.000000 s
        Salinity: Fresh
        CyclePeriod (calculated): 0.054997 s
